libs/color/colors.ts

```diff
@@ -0,0 +1,213 @@
+    }
+
+    /**
+    * Get the RGB value of a known color
+    */
+    //% blockId=colorscolors block="%color"
+    //% help="colors/well-known"
+    //% shim=TD_ID
+    //% weight=20 blockGap=8
+    //% blockHidden=true
+    export function wellKnown(color: Colors): number {
+        return color;
+    }
+
+    /**
+     * Convert an HSV (hue, saturation, value) color to RGB
+     * @param hue value of the hue channel between 0 and 255. eg: 255
+     * @param sat value of the saturation channel between 0 and 255. eg: 255
+     * @param val value of the value channel between 0 and 255. eg: 255
+     */
+
+    //% blockId="colorshsv" block="hue %hue|sat %sat|val %val"
+    //% hue.min=0 hue.max=255 sat.min=0 sat.max=255 val.min=0 val.max=255
+    //% help="colors/hsv"
+    //% weight=17
+    //% blockHidden=true
+    export function hsv(hue: number, sat: number = 255, val: number = 255): number {
+        let h = (hue % 255) >> 0;
+        if (h < 0) h += 255;
+        // scale down to 0..192
+        h = (h * 192 / 255) >> 0;
+
+        //reference: based on FastLED's hsv2rgb rainbow algorithm [https://github.com/FastLED/FastLED](MIT)
+        let invsat = 255 - sat;
+        let brightness_floor = ((val * invsat) / 255) >> 0;
+        let color_amplitude = val - brightness_floor;
+        let section = (h / 0x40) >> 0; // [0..2]
+        let offset = (h % 0x40) >> 0; // [0..63]
+
+        let rampup = offset;
+        let rampdown = (0x40 - 1) - offset;
+
+        let rampup_amp_adj = ((rampup * color_amplitude) / (255 / 4)) >> 0;
+        let rampdown_amp_adj = ((rampdown * color_amplitude) / (255 / 4)) >> 0;
+
+        let rampup_adj_with_floor = (rampup_amp_adj + brightness_floor);
+        let rampdown_adj_with_floor = (rampdown_amp_adj + brightness_floor);
+
+        let r: number;
+        let g: number;
+        let b: number;
+        if (section) {
+            if (section == 1) {
+                // section 1: 0x40..0x7F
+                r = brightness_floor;
+                g = rampdown_adj_with_floor;
+                b = rampup_adj_with_floor;
+            } else {
+                // section 2; 0x80..0xBF
+                r = rampup_adj_with_floor;
+                g = brightness_floor;
+                b = rampdown_adj_with_floor;
+            }
+        } else {
+            // section 0: 0x00..0x3F
+            r = rampdown_adj_with_floor;
+            g = rampup_adj_with_floor;
+            b = brightness_floor;
+        }
+        return rgb(r, g, b);
+    }
+
+    /**
+     * Fade the color by the brightness
+     * @param color color to fade
+     * @param brightness the amount of brightness to apply to the color, eg: 128
+     */
+    //% blockId="colorsfade" block="fade %color=neopixel_colors|by %brightness"
+    //% brightness.min=0 brightness.max=255
+    //% help="light/fade"
+    //% group="Color" weight=18 blockGap=8
+    //% blockHidden=true
+    export function fade(color: number, brightness: number): number {
```

Comment:
``fade`` would be overloaded a bit with my extension (which has fades between different color palettes), if this needs to stay the same name for backwards compat though I can figure something out / disambiguate somehow

Reply:
```
 export function fade(color: number, brightness: number, otherColor = 0): number 
```
?

Reply:
this is fade between 2 colors, not color buffers.

Reply:
Maybe I just change my package to have the class ``PaletteFade`` instead of ``Fade`` then when merging in, and rename the necessary functions, easy change and more clear what the class is for anyway